src/FindEmailAddress.java

```diff
@@ -0,0 +1,362 @@
+/*
+*
+* The project is for finding email address in the web site
+*
+* Create a command line program that will take an internet domain name (i.e. “jana.com”) and
+* print out a list of the email addresses that were found on that website only.
+* the application will find email addresses on any discoverable page of the website.
+*
+* We implemented a multithread method. each url request and email&href check is a threadTask
+*
+*
+*/
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Hashtable;
+import java.net.URI;
+import java.util.concurrent.*;
+
+/*
+*  This is the main class to get the email addresses
+*
+*/
+public class FindEmailAddress {
+
+    private String rootURL;
+    public String requestDomain;
+    public Hashtable<String,String> urlHistory;
+    public Hashtable<String,String> emails;
+    public static int maxUrlCheck = 200; //to avoid too many pages to check we add maximum number of url to check, it can be modified
+    public ExecutorService executor;
+    public Integer tasksNumber;
+
+    public static void main(String[] args)
+    {
+        String requestUrl = null;
+
+        if(args.length >0)
+        {
+            requestUrl =  args[0];
+        }
+        if (requestUrl == null)
+            return;
+
+        FindEmailAddress findEmailAddress = new FindEmailAddress(requestUrl);
+        findEmailAddress.startWebCrawl();
+        findEmailAddress.printout();
+    }
+
+    /*
+    *  @param requestURL is the root url. All the webpage with same domain will be searched for email address
+    */
+    public FindEmailAddress(String requestURL)
+    {
+        this.rootURL = requestURL;
+        if(!requestURL.toLowerCase().startsWith("http"))
+        {
+            rootURL = "http://" + requestURL;
+        }
+        try {
+            URI uri = new URI(rootURL);
+            this.requestDomain= uri.getHost();
+
+        }catch(java.net.URISyntaxException e)
+        {
+            e.printStackTrace();
```

Comment:
you're catching an exception here and printing a stack trace, but then the code is simply continuing on. I might just let the exception go here (don't catch it), but at the least I think you want to return rather than continue.

Reply:
Yes. should let exception go.